src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,188 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use std::{borrow::BorrowMut, cmp::Ordering};
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgGroup, Command};
+use pid::walk_pid;
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let default_delimiter = "\n".into();
+    let delimiter = matches
+        .get_one::<String>("delimiter")
+        .unwrap_or(&default_delimiter);
+
+    let mut programs = matches
+        .get_many::<String>("pattern")
+        .unwrap_or_default()
+        .map(|it| it.to_string())
+        .collect::<Vec<_>>();
+
+    let should_ignore_case = matches.get_flag("ignore-case");
+    let should_inverse = matches.get_flag("inverse");
+
+    let flag_newest = matches.get_flag("newest");
+    let flag_oldest = matches.get_flag("oldest");
+
+    if flag_newest != flag_oldest {
```

Comment:
I think the following block works fine for `pgrep -o` and `pgrep -n`. However, there is an additional use case with a pattern that's currently not covered:
```
$ pgrep konsole
1236
1657
1881
1914
1936
$ pgrep -o konsole
1236
$ pgrep -n konsole
1936

$ cargo run pgrep konsole
1236
1657
1881
1914
1936
$ cargo run pgrep -o konsole 
1
$ cargo run pgrep -n konsole 
8237
```

Reply:
I'll fix it later

Reply:
It works fine now.

```shell
❯ cargo run -p uu_pgrep -- -o zsh zsh
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
     Running `/home/krysztal/Desktop/project/uutils/procps/target/debug/pgrep -o zsh zsh`
pgrep: only one pattern can be provided
❯ cargo run -p uu_pgrep -- -o zsh
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
     Running `/home/krysztal/Desktop/project/uutils/procps/target/debug/pgrep -o zsh`
6174
```